src/Microsoft.VisualStudio.Services.Agent/ProcessInvoker.cs

```diff
@@ -119,69 +102,69 @@ private void Start(string workingDirectory, string fileName, string arguments, I
 
             // Hook up the events.
             _proc.EnableRaisingEvents = true;
-            _proc.Exited += OnExited;
-            if (_proc.StartInfo.RedirectStandardOutput)
-            {
-                _proc.OutputDataReceived += OnOutputDataReceived;
-            }
+            _proc.Exited += ProcessExitedHandler;
 
-            if (_proc.StartInfo.RedirectStandardError)
+            using (var registration = cancellationToken.Register(() => CancelProcessTree()))
             {
-                _proc.ErrorDataReceived += OnErrorDataReceived;
-            }
+                // Start the process.
+                try
+                {
+                    _stopWatch = Stopwatch.StartNew();
+                    _proc.Start();
+                }
+                catch (Exception ex) when (ex is Win32Exception && (ex as Win32Exception)?.NativeErrorCode == 2)
```

Comment:
Is this required anymore, or was this a side effect of creating a separate process start info. I remember some strange difference about the error message for the file name or working directory when the new instance was used instead of the default one.

It looks like this exception handler is only for Windows. What about OSX/Linux?